src/ui_kit/src/folder/mod.rs

```diff
@@ -63,42 +65,87 @@ pub fn Folder(cx: Scope<Props>) -> Element {
+                                        log::error!("Error on go back a directory: {error}");
+                                    };
+                                    update_current_dir.set(());
+                                    break;
+                                }
+                        
+
+                                if let Some(file_name) = drag_file_event_in_app.file_name {
+                                    *drag_over_folder.write_silent() = false;
+                                    let mut file_name = file_name.clone();
+                                    let current_directory = file_storage.current_directory().unwrap_or_default();  
+                                    let folder_name = folder_name_complete_ref.with(|name| name.clone());
+                                    let directory_target = match current_directory.get_item(&folder_name).and_then(|item| item.get_directory()) {
+                                            Ok(dir) => dir,
+                                            _ => return
+                                    };
+                                    let file = current_directory.get_item(&file_name).unwrap();
+
+                                    file_name = files_functions::verify_duplicate_name(directory_target.clone(), 
+                                    file.name().clone(), PathBuf::from(file.name()));
+                                    if file_name.ne(&file.name()) {
+                                       if let Err(error) =  file.rename(&file_name) {
+                                        log::error!("Error renaming file to move into another folder: {error}");
+                                        return;
+                                       }
+                                    }
+                                      
+                                        match directory_target.add_item(file.clone()) {
                                             Ok(_) => {
-                                                *drag_over_folder.write_silent() = false;
-                                                // TODO: Remove all files inside this folder
-                                                log::info!("file from directory was deleted.");
-                                            },
-                                            Err(error) => log::error!("Error deleting file from directory: {error}"),
-                                        }
-                                },
-                                    Err(error) => log::error!("Error adding file into directory: {error}"),
-                                };
-                                let file_leave_folder_js = include_str!("./file_leave_folder.js").replace("folder-id", &folder_id);
-                                eval_script.eval(&file_leave_folder_js);
-                            }
+                                                tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
Any reason for sleeping the task?

Reply:
Hmm almost sounds like using a notify would be a good solution here but this will be fine to use